orbit/cmd/desktop/desktop.go

```diff
@@ -60,6 +61,14 @@ func setupRunners() {
 }
 
 func main() {
+	// FIXME: we need to do a better job of graceful shutdown, releasing resources, stopping
+	// tickers, etc. (https://github.com/fleetdm/fleet/issues/21256)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer func() {
+		log.Info().Msg("canceling context")
+		cancel() // TODO: move this to onExit?
```

Comment:
based on what we saw today with the ticker, won't this be canceled immediately after main runs? I think we might need to move it to `onExit`

Reply:
I don't think this is quite the same as what we saw with the ticker. In this case, the defer is at `main` level. In the other case it was happening inside the `onReady` call.

So in this case, `systray.Run` is blocking the main defer until `onExit` is called. By contrast, the `onReady` function exits as soon as the various resources and goroutines are kicked off, which that defer gets called pretty much immediately after start up.

Reply:
I do think we should consolidate it into `onExit` or some other graceful shutdown function though.

Reply:
that makes sense 👍